controllers/tab/issueEntry/form/IssueEntryPublicationMetadataForm.inc.php

```diff
@@ -341,6 +344,21 @@ function execute($request) {
 				$submission->setLicenseURL(null);
 			}
 
+			// Versioning
+
+			// set publication date
+			$submission->setDatePublished($this->getData('datePublished') ? $this->getData('datePublished') : Core::getCurrentDate());
```

Comment:
Because this is happening here, and not above as earlier i.e. because it does not considers if an article is unassigned from an issue, an article can have the datePublished although it is not assigned to an issue i.e. although not published